### DISCUSSION TITLE
Support additional Graphs

### DISCUSSION THREAD

**Discussion by kg-intec** [JakubAndrysek/MkDoxy#43] [Ideas]
First, nice plugin 👍

However, good things can always be improved... There are many types of additional graphs supported by doxygen:

- Call graphs
- Caller Graphs
- UML diagrams
- Directory Graphs

As well as diagrams from PlantUML, which can be defined directly in the code. Support for this would be great!

Examples:

- [Class Graph](https://github.com/kg-intec/images/blob/main/class.png)
- [Directory Graph](https://github.com/kg-intec/images/blob/main/file.png)

**Comment by JakubAndrysek** [comment]
Thank you for a good idea, I will look at it in the future. 
Generating automatic navigation has got a higher priority.

**Comment by JakubAndrysek** [comment]
About Graphviz in Mkdocs
https://eskool.gitlab.io/mkhack3rs/graphviz/

About PlantUML in Mkdocs
https://eskool.gitlab.io/mkhack3rs/plantuml/

To enable this feature in Doxygen
- `HAVE_DOT = YES`
- https://www.doxygen.nl/manual/config.html#cfg_have_dot

Choose inheritance graph
- DOT
  - It is by default
  https://graphviz.org/doc/info/lang.html
- UML
  - It is optional and must be enabled in Doxyfile
  - `UML_LOOK = YES`
  - https://www.doxygen.nl/manual/config.html#cfg_uml_look


Other interesting options
- `GENERATE_LEGEND = NO`
  - https://www.doxygen.nl/manual/config.html#cfg_generate_legend
  - It will disable the legend in the bottom of the page

- `SEARCHENGINE = NO`
  - https://www.doxygen.nl/manual/config.html#cfg_searchengine
  - It will disable the search engine in the top of the page - speed up the page load

**Comment by JakubAndrysek** [comment]
It is not possible to generate any graphs withouth `GENERATE_HTML = YES` in Doxyfile.

The MkDoxy plugin will use:
- XML output
  - for recursive parsing of the Doxygen output

- HTML output
  - for reusing the images generated by Doxygen

Sadly it is probably not possible to generate source code for custom graphs rendering.
- Graphviz supports exporting images to SVG, but the result contains whit background (not transparent)

**Reply by amit-gueta** [reply]
Hi @JakubAndrysek 
does it already work with mkdoxy?  
I mean, I could generate the html folder into mkdoxy, but mkdocs serve doesn't show the graphs

thanks!

**Reply by JakubAndrysek** [reply]
Yes, it is as you described. Showing graphs in md pages is not implemented.
If you would like to help with development, I would be very grateful.

Have a nice day, Jacob

**Comment by JakubAndrysek** [comment]
Something is coming!

![image](https://github.com/JakubAndrysek/MkDoxy/assets/33494544/24bfb401-2c66-4d16-898c-ef363c59e044)

![image](https://github.com/JakubAndrysek/MkDoxy/assets/33494544/2969a61c-bf75-4b6e-a7f5-64caa6e30a16)

**Comment by JakubAndrysek** [comment]
Related issue: https://github.com/JakubAndrysek/MkDoxy/issues/83

**Comment by JakubAndrysek** [comment]
I have tried to implement showing diagrams, but it needs more time to finish the task.
The branch is under development: [diagrams-support](https://github.com/JakubAndrysek/MkDoxy/tree/diagrams-support?rgh-link-date=2023-12-27T14%3A38%3A21Z)

```yaml
GENERATE_HTML: YES
HAVE_DOT: YES
DOT_IMAGE_FORMAT: svg
```
![image](https://github.com/JakubAndrysek/MkDoxy/assets/33494544/6405d246-b539-4415-86ee-b41d210d6ff8)

<hr>

```yaml
GENERATE_HTML: YES
HAVE_DOT: YES
DOT_IMAGE_FORMAT: svg
UML_LOOK: YES
```
![image](https://github.com/JakubAndrysek/MkDoxy/assets/33494544/15e5ab0b-74da-4797-9473-f36393e0428d)

**Comment by MatthijsvanDelft** [comment]
Really like the plugin and are planning to use it a lot!:)
Having these UML diagrams would be really beneficial imo!
Are there any plans on when to release this?
Thanks in advance
